libdrgn/register_state.c

```diff
@@ -31,6 +31,27 @@ struct drgn_register_state *drgn_register_state_create_impl(uint32_t regs_size,
 	return regs;
 }
 
+struct drgn_register_state *drgn_register_state_dup(
+		const struct drgn_register_state *regs)
+{
+	uint32_t bitset_size = ((uint32_t)regs->num_regs + CHAR_BIT + 1) / CHAR_BIT;
+	size_t size = 0;
+	struct drgn_register_state *ret;
+	if (__builtin_add_overflow(regs->regs_size, bitset_size, &size) ||
+	    __builtin_add_overflow(size, sizeof(*regs), &size) ||
+	    !(ret = malloc(size)))
+		return NULL;
+
+	ret->module = regs->module;
+	ret->regs_size = regs->regs_size;
+	ret->num_regs = regs->num_regs;
+	ret->interrupted = regs->interrupted;
+	ret->_pc = regs->_pc;
+	ret->_cfa = regs->_cfa;
+	memcpy(ret->buf, regs->buf, regs->regs_size);
```

Comment:
This can be a single `memcpy`:

```suggestion
	memcpy(ret, regs, size);
```